ProcessQueue/__init__.py

```diff
@@ -159,320 +190,343 @@ def main(msg: func.QueueMessage):
+                        )
+                        concept_id = concept_id["concept_id"]
+                    except:
+                        concept_id = -1
+                else:
+                    concept_id = -1
 
-            # POST the ScanReportConcept data to the model
-            response = requests.post(
-                url=api_url + "scanreportconcepts/",
+            else:
+                val_desc = None
+                concept_id = -1
+
+            # Create ScanReportValue entry
+            # We temporarily utilise the redundant 'conceptID' field in ScanReportValue
+            # to save any looked up conceptIDs in the previous block of code.
+            # The conceptID will be cleared later
+            scan_report_value_entry = {
+                "created_at": datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+                "updated_at": datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+                "value": value,
+                "frequency": int(frequency),
+                "conceptID": concept_id,
+                "value_description": val_desc,
+                "scan_report_field": names_to_ids_dict[name],
+            }
+
+            # Append to list
+            value_entries_to_post.append(scan_report_value_entry)
+
+        print("POST values", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
+        # POST value_entries_to_post to ScanReportValues model
+        values_response = requests.post(
+            url=api_url + "scanreportvalues/", 
+            data=json.dumps(value_entries_to_post), 
+            headers=headers
+        )
+        if values_response.status_code != 201:
+            raise HTTPError(' '.join(['Error in values save:', str(values_response.status_code), str(json.dumps(value_entries_to_post))]))
+
+        print("POST values finished", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
+
+        # Process conceptIDs in ScanReportValues
+        # GET values where the conceptID != -1 (i.e. we've converted a concept code to conceptID in the previous code)
+        print("GET posted values", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
+        get_ids_of_posted_values = requests.get(
+            url=api_url
+            + "scanreportvaluepks/?scan_report="
+            + str(body["scan_report_id"]),
+            headers=headers,
+        )
+        print("GET posted values finished", datetime.utcnow().strftime("%H:%M:%S.%fZ"))
+
+        ids_of_posted_values = json.loads(get_ids_of_posted_values.content.decode("utf-8"))
+
+        # Create a list for a bulk data upload to the ScanReportConcept model
+
+        concept_id_data = [{
+                "nlp_entity": None,
+                "nlp_entity_type": None,
+                "nlp_confidence": None,
+                "nlp_vocabulary": None,
+                "nlp_processed_string": None,
+                "concept": concept["conceptID"],
+                "object_id": concept["id"],
+                "content_type": 17,

```

Comment:
`17` seams really random, i guess it is. Is it safe to assume it's the same ID across all databases (I think so), or would be the same if a completely new instance of this webtool was started?

https://stackoverflow.com/questions/12716970/get-contenttype-id-in-django-for-generic-relation
like:
```
"content_type": ContentType.objects.get(model='scanreportvalue').id
```

Reply:
ahhhh of course